src/rgw/rgw_kms.cc

```diff
@@ -41,7 +41,7 @@ class ZeroPoolAllocator {
     struct element {
 	struct element *next;
 	int size;
-	char data[4];
+	char data[32];
```

Comment:
From what I understood, this allocator is designed to store variable-length payloads inline after `element`:
```cpp
element *nw { (element *) malloc(sizeof *b + ns) };
```
Perhaps declaring data with a fixed size is what’s triggering GCC’s false positive in the first place?
```suggestion
	char data[];
```